src/piano/Recorder.java

```diff
@@ -0,0 +1,49 @@
+package piano;
+
+import java.util.ArrayList;
+
+public class Recorder
+{
+    private ArrayList<Key> recordedKeys = new ArrayList<>();
```

Comment:
Instead of creating parallel arrays, make one object that holds both the key and the time it was pressed.